### ISSUE TITLE
cx-icon needs to have aria-label

### ISSUE THREAD

**Issue by saad-mo** [SAP/spartacus#11317]
Update `cx-icon` to have `aria-label` attribute. This attribute should be required to prevent empty labels. May be a non-breaking change :orange_heart:

- [ ] [Check Full Implementation details ](https://cxwiki.sap.com/display/spar/Text+Alternative+for+Non-Text+Content)
- [ ] Update  (asterix or praetorians)
- [ ] Fix SCSS (praetorians) 
- [ ] Fix Unit tests (asterix)
- [ ] Fix e2e test (asterix)

**Comment by saad-mo** [comment]
Solutions for this is provided in a [related ticket here]( https://github.com/SAP/spartacus/issues/13438#issuecomment-933867702) 

This ticket is also very generic and does not specify where the issue is happening. Similar problems are covered in other tickets so closing this for now.
